lib/heap_sort.rb

```diff
@@ -1,8 +1,20 @@
-
+require_relative('min_heap')
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
-def heap_sort(list)
-  raise NotImplementedError, "Method not implemented yet..."
+# Time Complexity:  O(n log n) where n is the number of items in the list
+# Space Complexity: O(log n)
+def heapsort(list)
```

Comment:
Since your'e building a full heap of n nodes... this has space complexity of O(n).  There is a way to use the original array as the internal array for the heap and thus keep it at O(1) space complexity.  Oh and you're also creating a new sorted array.

Otherwise this works.